packages/ifpack2/src/Ifpack2_PowerMethod.hpp

```diff
@@ -152,70 +155,142 @@ powerMethodWithInitGuess (const OperatorType& A,
      "compute the initial guess), or because the norm2 method has a "
      "bug.  The first is not impossible, but unlikely.");
 
-  if (verbose) {
-    *out << "  Original norm1(x): " << x->norm1 ()
+  if(verbose) {
+    *out << "  Original norm1(x): " << x->norm1()
           << ", norm2(x): " << norm << endl;
   }
 
-  x->scale (one / norm);
+  x->scale(one / norm);
 
-  if (verbose) {
-    *out << "  norm1(x.scale(one/norm)): " << x->norm1 () << endl;
+  if(verbose) {
+    *out << "  norm1(x.scale(one/norm)): " << x->norm1() << endl;
   }
 
   if(y.is_null())
-    y = Teuchos::rcp(new V(A.getRangeMap ()));
-
-  for (int iter = 0; iter < numIters-1; ++iter) {
-    if (verbose) {
-      *out << "  Iteration " << iter << endl;
+    y = Teuchos::rcp(new V(A.getRangeMap()));
+
+  // GH 04 Nov 2022: The previous implementation of the power method was incorrect.
+  // It computed x->norm2() inside the loop, but returned x^T*Ax. This returned horribly
+  // incorrect answers in certain cases, such as the Cheby_belos test, which has two 
+  // largest eigenvalues of 12.5839 and -10.6639. The power method iteration history 
```

Comment:
The answers were correct, just not consistent with the debug output.

Reply:
It some cases it was reporting negative leading eigenvalues for a matrix with a leading positive eigenvalue, which you said yourself was definitely wrong :P

Reply:
No, it wasn't. It was the correct estimate given the prescribed number of iterations.

Reply:
Okay I see what you mean. I updated the comment accordingly.